### PR TITLE
 Add support for `Postgresql schema` parameter 

### PR DESCRIPTION
**Pull request by grihabor**
Hi,

I need to store `schema_version` in different postgres schema than `public`, so I've introduced `-m` or `--schema` cli parameter to manually set the schema of `schema_version` relation. I need your review, thanks!

### CODE REVIEW

File: pgmigrate.py [hunk 1]
```diff
@@ -494,15 +502,16 @@ def _get_callbacks(callbacks, base_dir=''):
     return _parse_str_callbacks(callbacks, ret, base_dir)
 
 
-def _migrate_step(state, callbacks, base_dir, user, cursor):
+def _migrate_step(state, callbacks, base_dir, user, db_handle):
```
Comment (grihabor) [inline pgmigrate.py:505]: `db_handle=(cursor, schema)` is a workaround to make pylint `too-many-arguments` check pass

**Comment by secwall** [comment]
This change will make life more difficult. Imagine the following situation:

We have schemas A, B, and C.
We perform migration 1 which adds table T1 to schema A with `IF NOT EXISTS` clause (probably not so good idea anyway, but still there are a lot of users doing such things) and store migration info in schema B.
Then we perform migration 2 which adds table T2 to schema A and store migration info in schema C.
Now if we run pgmigrate info with schema B it'll tell us that our version is 1 which is totally wrong.
And if we run pgmigrate info on schema C it'll tell us that our version is 2.

This example is too simplified but there are a lot of not so trivial cases with migrations with DML.
Basic idea is simple: you should migrate your whole database as one and storing migration version in one place will help us avoid some really messy situations.

**Comment by grihabor** [comment]
This patch is intended to add flexibility for users who need it and who knows what they are doing.

I think the problem you describe is totally the user's fault. If one decides to store the `version_info` in a different table than `public`, he must understand all the consequences.

**Comment by grihabor** [comment]
The default value for the parameter is still `public`, so the user can decide to stick to the default and avoid the complexity.

**Pull request closed without merging**